src/Lib/Router.php

```diff
@@ -195,8 +196,32 @@ public static function isCallable($obj, $method, $inheritsFrom = 'Controller')
                 return false;
             }
 
+            $reflectedMethod = false;
             try {
                 $reflectedMethod = $reflectedObj->getMethod($method);
+            } catch (ReflectionException $e) {
+                // The method does not exist on the object, but could still be available, e.g., via __call
+            }
+
+            // If the method does not exist, but is still callable (e.g. via __call), then only ensure it meets
+            // the same criteria on inheritance and public uncallables
+            if ($reflectedMethod === false) {
+                try {
+                    return (
+                        is_callable([$obj, $method])
+                        && !in_array(strtolower($method), $publicUncallables)
+                        && (
+                            $inheritsFrom === null
+                            || $reflectedObj->getName() == $inheritsFrom
+                            || $reflectedObj->isSubclassOf($inheritsFrom)
+                        )
+                    );
+                } catch (Exception $e) {
```

Comment:
What Exception instance is possible here? Should probably catch the most specific avalable.